containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/org/hawkular/openshift/namespace/NamespaceOverrideMapper.java

```diff
@@ -0,0 +1,271 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.openshift.auth.org.hawkular.openshift.namespace;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+
+import org.jboss.logging.Logger;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.JdkSSLOptions;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.SSLOptions;
+import com.datastax.driver.core.Session;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ *
+ */
+public class NamespaceOverrideMapper {
+
+    private static final Logger log = Logger.getLogger(NamespaceOverrideMapper.class);
+
+    private Session session;
+
+    private final String token;
+
+    public Map<String,String> overrides;
+
+    public NamespaceOverrideMapper(String token) {
+        log.info("Creating NamespaceOverrideMapper");
+
+        this.token = token;
+
+        boolean connected = false;
+        do {
+            try {
+                session = createSession();
+                connected = true;
+            } catch (Exception e) {
+                log.error("Could not connect to Cassandra. This could mean Cassandra is not yet up and running. Will try again.");
+                e.printStackTrace();
+                try {
+                    Thread.sleep(2000);
+                } catch (InterruptedException e1) {
+                    e1.printStackTrace();
+                }
+            }
+        } while (!connected);
+        log.info("Connection to Cassandra establised");
+
+        // ensure that the openshift_metrics keyspace exists
+        ensureKeyspace("openshift_metrics");
+
+        // if we already have a mappingTable stored in OpenShift, then use it.
+        // Otherwise we need to generate the overwrite table.
+        if (hasMappingTable()) {
+            log.info("Mapping table already exists in Cassandra. Using this table");
+            this.overrides = getMappingTable();
+        } else {
+            log.info("Mapping table does not exist in Cassandra. Creating Mapping");
+            // get the current list of tenants in Hawkular Metrics
+            Set<String> tenants = getTenants();
```

Comment:
Is it guaranteed that this runs *after* the Hawkular-Metrics schema has been created?